homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -1,174 +1,202 @@
-        self.update_interval = conf[CONF_SCAN_INTERVAL]
-        self.show_on_map = conf[CONF_SHOW_ON_MAP]
-        self._hass = hass
-
-    def setup(self, latitude, longitude, radius, additional_stations):
-        """Set up the tankerkoenig API.
 
-        Read the initial data from the server, to initialize the list of fuel stations to monitor.
-        """
-        _LOGGER.debug("Fetching data for (%s, %s) rad: %s", latitude, longitude, radius)
-        try:
-            data = pytankerkoenig.getNearbyStations(
-                self._api_key, latitude, longitude, radius, "all", "dist"
-            )
-        except pytankerkoenig.customException as err:
-            data = {"ok": False, "message": err, "exception": True}
-        _LOGGER.debug("Received data: %s", data)
-        if not data["ok"]:
-            _LOGGER.error(
-                "Setup for sensors was unsuccessful. Error occurred while fetching data from tankerkoenig.de: %s",
-                data["message"],
-            )
-            return False
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            name=name,
+            update_interval=timedelta(minutes=update_interval),
+        )
 
-        # Add stations found via location + radius
-        if not (nearby_stations := data["stations"]):
-            if not additional_stations:
-                _LOGGER.error(
-                    "Could not find any station in range."
-                    "Try with a bigger radius or manually specify stations in additional_stations"
-                )
-                return False
-            _LOGGER.warning(
-                "Could not find any station in range. Will only use manually specified stations"
-            )
-        else:
-            for station in nearby_stations:
-                self.add_station(station)
+        self._api_key = entry.data[CONF_API_KEY]
+        self._selected_stations = entry.data[CONF_STATIONS]
+        self._hass = hass
+        self.stations: dict[str, dict] = {}
+        self.fuel_types = entry.data[CONF_FUEL_TYPES]
+        self.show_on_map = entry.options[CONF_SHOW_ON_MAP]
 
-        # Add manually specified additional stations
-        for station_id in additional_stations:
+    def setup(self):
+        """Set up the tankerkoenig API."""
+        for station_id in self._selected_stations:
             try:
-                additional_station_data = pytankerkoenig.getStationData(
-                    self._api_key, station_id
-                )
+                station_data = pytankerkoenig.getStationData(self._api_key, station_id)
             except pytankerkoenig.customException as err:
-                additional_station_data = {
+                station_data = {
                     "ok": False,
                     "message": err,
                     "exception": True,
                 }
 
-            if not additional_station_data["ok"]:
+            if not station_data["ok"]:
```

Comment:
What would be the reason for this case? How can the user resolve it?

Reply:
the `return False` was left from prior code ... will changes this to just skip a failing station (_eq. an imported or just a station_id which does not exit anymore_)

Reply:
ah, now I get your thoughts ... there is no catch for a possible "Auth failed" situation, yet. But I agree, this needs to be added soon, so that the re-configuration can be triggered

Reply:
Yes, good to do it separately. 👍

Reply:
- #72682